trade-finance-manager-ui/server/controllers/case/amendments/amendmentFacilityEndDate.controller.js

```diff
@@ -38,13 +38,28 @@ const getAmendmentFacilityEndDate = async (req, res) => {
   const facilityEndDateMonth = facilityEndDate ? format(new Date(facilityEndDate), 'M') : '';
   const facilityEndDateYear = facilityEndDate ? format(new Date(facilityEndDate), 'yyyy') : '';
 
+  const facility = await api.getFacility(facilityId, userToken);
```

Comment:
I've just tried adding the following docstring: 

```
/**
 * @param id - The facility ID
 * @param token - The user token
 * @returns {Promise<{ status: number, data: string }> | Promise<import('../../libs/common/src/types/mongo-db-models/tfm-facilities').TfmFacility> | {}}
 */
```

(bit of a messy return type), but it seems unhappy back in the `amendmentFacilityEndDate` controller in the line `facility?.facilitySnapshot?.dates?.facilityEndDate`, even with the optional chaining - specifically it seems unhappy about referencing `facilitySnapshot` on it.

Reply:
Ah yes that's my bad - I thought this was in `tfm-api` not `tfm-ui`. The `TfmFacility['facilitySnapshot']` object does not have the `dates` object on it because `tfm-api` performs mapping to the `TfmFacility` returned from `dtfs-central-api`

Reply:
I realise my previous comment doesn't actually resolve this thread. I think in this case, adding a docstring to this api method would be quite hard to do as documenting the return type of `api.getFacility` would probably take a lot of extra work, so I'm going to resolve this thread